File: src/main/java/dev/fincke/hopper/platforms/PlatformRepository.java
package dev.fincke.hopper.platforms;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface PlatformRepository extends JpaRepository<Platform, UUID> 
{
    
    List<Platform> findByPlatformType(String platformType);
    
    Platform findByName(String name);
}

File: src/main/kotlin/dev/fincke/hopper/platforms/PlatformRepository.kt
package dev.fincke.hopper.platforms

import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

// data access layer for Platform (auto implements CRUD & query methods)
interface PlatformRepository : JpaRepository<Platform, UUID>
{
    // finds platform by type
    fun findByPlatformType(platformType: String): List<Platform>
    // finds platform by name
    fun findByName(name: String): Platform?
}
